src/RealtimeEffectPanel.cpp

```diff
@@ -1000,6 +1000,77 @@ RealtimeEffectPanel::RealtimeEffectPanel(
             break;
          }
    });
+
+   mUndoSubscription = UndoManager::Get(mProject).Subscribe(
+      [this](UndoRedoMessage message)
+      {
+         if (
+            message.type == UndoRedoMessage::Type::Purge ||
+            message.type == UndoRedoMessage::Type::BeginPurge ||
+            message.type == UndoRedoMessage::Type::EndPurge)
+            return;
+
+         auto& trackList = TrackList::Get(mProject);
+
+         // Realtime effect UI is only updated on Undo or Redo
+         auto waveTracks = trackList.Any<WaveTrack>();
+         
+         if (message.type == UndoRedoMessage::Type::UndoOrRedo)
```

Comment:
You don't unpdate UI for the Reset message, which can happen if you choose a state using the History dialog, and so the title of a dialog might be incorrect in that case.  However that case will correctly fall through to the check for reachability of states below.